#  Copyright 2024 Dataport. All rights reserved. Developed as part of the MERLOT project.
#
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.

diff --git a/fc-service-core/src/main/java/eu/xfsc/fc/core/service/schemastore/SchemaStoreImpl.java b/fc-service-core/src/main/java/eu/xfsc/fc/core/service/schemastore/SchemaStoreImpl.java
index 05ad1b35..4d643f66 100644
--- a/fc-service-core/src/main/java/eu/xfsc/fc/core/service/schemastore/SchemaStoreImpl.java
+++ b/fc-service-core/src/main/java/eu/xfsc/fc/core/service/schemastore/SchemaStoreImpl.java
@@ -69,6 +69,10 @@ public class SchemaStoreImpl implements SchemaStore {
   @Override
   public int initializeDefaultSchemas() {
     log.debug("initializeDefaultSchemas.enter");
+
+    // Always start fresh
+    clear();
+
     int count = 0;
     int found = dao.getSchemaCount();
     if (found == 0) {
